rig-core/src/lib.rs

```diff
@@ -98,6 +98,7 @@ pub mod streaming;
 pub mod tool;
 pub mod transcription;
 pub mod vector_store;
+pub use serde_json;
```

Comment:
I'd rather we don't re-export `serde_json` here since that could create a whole lot of issues down the line

Reply:
Would you mind clarifying? Earlier we discussed re-exporting from `rig-core` so that the generated code would not require the user to have `serde_json` in their Cargo.toml. I removed schemars entirely, and I simlified the `call` impl to not need serde_json, but unfortunately `ToolDefinition` requires that its parameters field be a serde_json::value::Value. Should I just re-export from `rig-core-derive` instead?

Reply:
Yeah apologies, you're right that I'm contradicting myself here. But after thinking and reading about it (here is a small [blogpost](https://brokenco.de/2023/07/26/rust-re-export.html) talking about when to re-export and when not to), I'm not convinced that `serde_json` should be re-exported from any of our crates. If the user wants to use the macro, they would just need to also install `serde_json`.

If this becomes too much of a hassle we can always add the re-export later on. But adding it now and removing it would break the API, which we're trying to avoid as much as possible even if the crate is still in pre-1.0.0.

Reply:
Makes sense! `serde_json` is so common that it has a higher likelihood of creating those version mismatches.